client/data/newsletter-categories/use-set-newsletter-category-mutation.tsx

```diff
@@ -0,0 +1,47 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import request from 'wpcom-proxy-request';
+
+type useSetNewsletterCategoryMutationParams = {
+	siteId: number;
+};
+
+type SetNewsletterCategoryParams = {
+	id: number;
+};
+
+type SetNewsletterCategoryResponse = {
+	success: boolean;
+};
+
+const useSetNewsletterCategoryMutation = ( { siteId }: useSetNewsletterCategoryMutationParams ) => {
```

Comment:
Wouldn't it be easier to do just `( siteId: number)`  here?

Reply:
Additional thought: the endpoint can also take a string as siteId. This is just nitpicking, since this won't break on compile-time.

Reply:
I like the idea, I'll change it 👍